### ISSUE TITLE
More issues with field names with dots in them

### ISSUE THREAD

**Issue by mpalmer** [looplab/logspout-logstash#24]
Certain ill-mannered programs (I'm looking at _you_, `docker-registry`) emit JSON-formatted log entries, but use dot-delimited nesting in the key names, which [breaks with ES 2.x](https://github.com/looplab/logspout-logstash/issues/11).  Since the JSON emitted by these programs isn't necessarily intended to be consumed by logstash, it's not necessarily their _fault_ that they're producing logstash-incompatible JSON, but is instead logspout-logstash's responsibility to transmute the error-producing JSON into something that logstash will understand.

I can think of two options:
1. Replace all dots in keys with underscores (quick, dirty, and within my extremely limited Go skills to implement); or
2. Translate dot-encoded keys into a nested structure (far superior, but there's no way known I can implement that all by myself).

Would a PR to implement option 1 be merged, or should I wait for someone with better Go chops to implement 2?

**Comment by maxekman** [comment]
Hey, sorry for not responding sooner. Is this still an issue for you? Which solution would you prefer as a fix in that case?

**Comment by mpalmer** [comment]
Yes, it's still an issue.  Option 2 would be a better outcome, by far, because it maintains structure, but someone else would have to implement it because I've got near-zero Go skills.  I could do a PR for option 1, though, if nobody else has the time to code something better.
